sky/shell/platform/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -23,163 +22,163 @@
 
-  uint32_t id_ = kUninitializedNodeId;
-  std::vector<scoped_refptr<Node>> children_;
-  Node* parent_ = nullptr;
-
- private:
-  friend class base::RefCounted<Node>;
-
-  ~Node();
+  semantics::SemanticFlagsPtr _flags;
+  semantics::SemanticStringsPtr _strings;
+  Geometry _geometry;
+}
 
-  void ValidateGlobalRect();
-  void ValidateGlobalTransform();
+#pragma mark - Override base class designated initializers
 
-  AccessibilityBridge* bridge_;
+// Method declared as unavailable in the interface
+- (instancetype)init {
+  [self release];
+  [super doesNotRecognizeSelector:_cmd];
+  return nil;
+}
 
-  semantics::SemanticFlagsPtr flags_;
-  semantics::SemanticStringsPtr strings_;
-  Geometry geometry_;
+#pragma mark - Designated initializers
 
-  std::unique_ptr<SkMatrix44> global_transform_;
-  std::unique_ptr<SkRect> global_rect_;
+- (instancetype)initWithBridge:(sky::shell::AccessibilityBridge*)bridge
+                           uid:(uint32_t)uid {
+  DCHECK(bridge != nil) << "bridge must be set";
+  DCHECK(uid >= RootNodeId);
+  self = [super init];
 
-  DISALLOW_COPY_AND_ASSIGN(Node);
-};
+  if (self) {
+    _bridge = bridge;
+    _uid = uid;
+  }
 
-AccessibilityBridge::Node::Node(AccessibilityBridge* bridge,
-                                const semantics::SemanticsNodePtr& node)
-    : bridge_(bridge) {
-  Update(node);
+  return self;
 }
 
-void AccessibilityBridge::Node::Update(
-    const semantics::SemanticsNodePtr& node) {
-  if (id_ == kUninitializedNodeId) {
-    id_ = node->id;
-  }
-  DCHECK(id_ == node->id);
+#pragma mark - Semantics node methods
+
+- (void)update:(const semantics::SemanticsNodePtr&)mojoNode {
+  DCHECK(_uid == mojoNode->id);
 
-  if (!node->flags.is_null()) {
-    flags_ = node->flags.Pass();
+  if (!mojoNode->flags.is_null()) {
+    _flags = mojoNode->flags.Pass();
   }
 
-  if (!node->strings.is_null()) {
-    strings_ = node->strings.Pass();
+  if (!mojoNode->strings.is_null()) {
+    _strings = mojoNode->strings.Pass();
   }
 
-  if (!node->geometry.is_null()) {
-    geometry_ = node->geometry.Pass();
+  if (!mojoNode->geometry.is_null()) {
+    _geometry = mojoNode->geometry;
   }
 
-  if (!node->children.is_null()) {
-    // Mark children for removal
-    for (scoped_refptr<Node> child : children_) {
-      DCHECK(child->parent_ != nullptr);
-      child->parent_ = nullptr;
+  if (!mojoNode->children.is_null()) {
+    // Mark existing children as orphans
+    NSArray* oldChildren = _children;
+    for (AccessibilityNode* child in oldChildren) {
+      DCHECK(child->_parent != nil);
+      child->_parent = nil;
     }
 
     // Set the new list of children
-    std::vector<scoped_refptr<Node>> children;
-    for (const semantics::SemanticsNodePtr& childNode : node->children) {
-      scoped_refptr<Node> child = bridge_->UpdateNode(childNode);
-      child->parent_ = this;
-      children.push_back(child);
+    NSMutableArray* children = [[NSMutableArray alloc] init];
+    _children = children;
```

Comment:
How many times can `update:` be called. If its more than once, you are leaking the old `_children`.